src/proxy/MultiMapProxy.ts

```diff
@@ -0,0 +1,204 @@
+import {MultiMapAddEntryListenerToKeyCodec} from './../codec/MultiMapAddEntryListenerToKeyCodec';
+import {MultiMapAddEntryListenerCodec} from './../codec/MultiMapAddEntryListenerCodec';
+import {MultiMapRemoveEntryListenerCodec} from './../codec/MultiMapRemoveEntryListenerCodec';
+import {MultiMapLockCodec} from '../codec/MultiMapLockCodec';
+import {MultiMapIsLockedCodec} from '../codec/MultiMapIsLockedCodec';
+import {MultiMapTryLockCodec} from '../codec/MultiMapTryLockCodec';
+import {MultiMapUnlockCodec} from '../codec/MultiMapUnlockCodec';
+import {MultiMapForceUnlockCodec} from '../codec/MultiMapForceUnlockCodec';
+
+export class MultiMapProxy<K, V> extends BaseProxy implements MultiMap<K, V> {
+
+    put(key: K, value: V): Promise<boolean> {
+        var keyData = this.toData(key);
+        var valueData = this.toData(value);
+        return this.encodeInvokeOnKey<boolean>(MultiMapPutCodec, keyData, keyData, valueData, 1);
+    }
+
+    get(key: K): Promise<Array<V>> {
+        var keyData = this.toData(key);
+
+        var deserializeCallback = (items: Array<Data>) => {
+            return items.map((item) => {
+                return this.toObject(item);
+            });
+        };
+
+        return this.encodeInvokeOnKey<Array<Data>>(MultiMapGetCodec, keyData, keyData, 1)
+            .then<Array<V>>(deserializeCallback);
+    }
+
+    remove(key: K, value: V): Promise<boolean> {
+        var keyData = this.toData(key);
+        var valueData = this.toData(value);
+        return this.encodeInvokeOnKey<boolean>(MultiMapRemoveEntryCodec, keyData, keyData, valueData, 1);
+    }
+
+    removeAll(key: K): Promise<Array<V>> {
+        var keyData = this.toData(key);
+        return this.encodeInvokeOnKey<Array<Data>>(MultiMapRemoveCodec, keyData, keyData, 1)
+            .then<Array<V>>((items: Array<Data>) => {
+                return items.map((item) => {
+                    return this.toObject(item);
+                });
+            });
+    }
+
+    keySet(): Promise<Array<K>> {
+        return this.encodeInvokeOnRandomTarget<Array<Data>>(MultiMapKeySetCodec)
+            .then<Array<K>>((items: Array<Data>) => {
+                return items.map((item) => {
+                    return this.toObject(item);
+                });
+            });
+    }
+
+    values(): Promise<Array<V>> {
+        return this.encodeInvokeOnRandomTarget<Array<Data>>(MultiMapValuesCodec)
+            .then<Array<V>>((items: Array<Data>) => {
+                return items.map((item) => {
+                    return this.toObject(item);
+                });
+            });
+    }
+
+    entrySet(): Promise<Array<[K, V]>> {
+        return this.encodeInvokeOnRandomTarget<Array<[Data, Data]>>(MultiMapEntrySetCodec)
+            .then<Array<[K, V]>>((entrySet: [Data, Data][]) => {
+                return entrySet.map((entry: Array<any>) => {
+                    var tuple: [K, V] = [this.toObject(entry[0]), this.toObject(entry[1])];
```

Comment:
you can still use `return <[K, V]>[this.toObject(entry[0]), this.toObject(entry[1])];`